include/zephyr/tracing/tracing.h

```diff
@@ -2376,6 +2376,65 @@
 
 /** @} */ /* end of subsys_tracing_apis_socket */
 
+/**
+ * @brief GPIO Tracing APIs
+ * @defgroup subsys_tracing_apis_gpio GPIO Tracing APIs
+ * @{
+ */
+
+/**
+ * @brief Trace a GPIO pin becoming active
+ * @param port @ref device representing the GPIO port
+ * @param pin @ref gpio_pin_t representing the pin
+ */
+#define sys_port_trace_gpio_pin_active(port, pin)
+
+/**
+ * @brief Trace a GPIO pin becoming inactive
+ * @param port @ref device representing the GPIO port
+ * @param pin @ref gpio_pin_t representing the pin
+ */
+#define sys_port_trace_gpio_pin_inactive(port, pin)
+
+/**
+ * @brief Trace GPIO being configured as output
+ * @param port @ref device representing the GPIO port
+ * @param pin @ref gpio_pin_t representing the pin
+ * @param flags @ref gpio_flags_t representing the flags
+ */
+#define sys_port_trace_gpio_pin_configured_output(port, pin, flags)
```

Comment:
Since you are passing the flags here, I don't see a reason to split this trace into two traces (`sys_port_trace_gpio_pin_configured_output` and `sys_port_trace_gpio_pin_configured_input`) since the output/input config is already encoded in the flags.

Reply:
This should be addressed. Please unresolve if you think it is not.